### PR TITLE
service/dap: fix race in no debug test

### PR DESCRIPTION
**Pull request by suzmue**
Close the exited channel first before sending status events.  This way when we disconnect after all the status messages, we know for sure the exit signal has been propagated and shutdown won't try to kill the process.

Updates go-delve/delve#2763

### CODE REVIEW

File: service/dap/server_test.go
Comment (polinasok) [inline service/dap/server_test.go:5034]: Kill-after-undetected-exit doesn't hurt, does it? So the only problem is inconsistent messages?
What about closing the channel before issuing process existed message and the terminated event? Then this case won't flake. And the case where the user clicks disconnect after process exists, but before everything is reflected is very unlikely.

Comment (suzmue) [reply service/dap/server_test.go:5034]: Done.

**Comment by polinasok** [comment]
I am not sure how I missed when this PR got updated and why the approval fell through the cracks. Approving now. 
I think the new version does the right thing to fix the race and should fix the flakiness that appears to be more common now.

- First close the exited channel
- Then send a terminated event
- This will trigger a disconnect request from the client
- The handler will now find a reliably closed exited channel (so no more flaky "terminating" output event)
- and just send a disconnect response back

**Review by polinasok — APPROVED** [review]
LGTM

**Pull request merged**
